### PR TITLE
kernel updates: switch to 5.10, upstream queue enabling

### PR DESCRIPTION
**Pull request by jan-kiszka**
Needs more testing and a double-check that we do not miss anything, both from our old queue as well as from upstream and the TI backlog.

Also: half-duplex support is lost again. But this is inherited from the SDK kernel where the related patches are coming from.

### CODE REVIEW

**Comment by AsuraZeng** [comment]
Hi Jan

I have a question that is not very clear. Do we have to use the upstream kernel？

Overall, many upstream and downstream porting are a great challenge to our system stability. I think we may need more time to test.

At present, I think using SDK 8.0 may be a better choice. Relatively speaking, the risk may be lower.

**Comment by jan-kiszka** [comment]
As we would have to test the SDK-based kernel as well, we could directly start with one that is under our control. Will also simplify updates. And give us more control over what we integrate (tones of unrelated TI SoC patches are skipped here).

While the number of patches seem overwhelming (and is sure not nice), most are orthogonal to the base kernel and were no effort to apply. Also -rt integrated smoothly. The highest risk is in selecting all needed patches. But that is no higher risk than building the 4.19 transition kernel.

If you feel unsure, you can also port the queue over the SDK kernel and test that one as well. That would allow us to base the decision on more data.

**Comment by BaochengSu** [comment]
LGTM, except it seems current we could have a single whole def_config file?

And for the 100 patches, could we tell which part is from TI kernel, and which part is backporting from upstream 5.14 and which part is our own wait-for-upstream ones? Having this info could simplify our maintainence work in the future, IMHO.

**Comment by jan-kiszka** [comment]
I didn't introduce the config split back then, so I can only speculate: I suppose the idea of the base defconfig was to provide that as BSP, for external layers that do not want the example bits and enablings.

Regarding queue structure: It's sorted already. Backports come first, then a mixture of TI SDK patches, pending upstream changes and own backlog. That part starts with `0044-watchdog-Start-watchdog-in-watchdog_set_last_hw_keep.patch`. I could tag the backports with `[upstream commit xxx]`-  which would also help if we propose them for 5.10-cip and move our base over, reducing the queue.

**Comment by jan-kiszka** [comment]
Just noticed that I'm missing one backport: mmc alias support.

**Comment by BaochengSu** [comment]
> I didn't introduce the config split back then, so I can only speculate: I suppose the idea of the base defconfig was to provide that as BSP, for external layers that do not want the example bits and enablings.
> 
> Regarding queue structure: It's sorted already. Backports come first, then a mixture of TI SDK patches, pending upstream changes and own backlog. That part starts with `0044-watchdog-Start-watchdog-in-watchdog_set_last_hw_keep.patch`. I could tag the backports with `[upstream commit xxx]`- which would also help if we propose them for 5.10-cip and move our base over, reducing the queue.

The config splitted was introduced to easy the ti kernel catchup pain, the original idea was to put the config options inherited from the TI sdk kernel to the base config and put the modifications that we made to the extra. So everytime when doing catchup, just to update the base config and keep the extra config as it is. Since from this moment we are moving to the upstream kernel, we could merge the two configs to a single one.

However, the use case you mentioned could be another helpful to users, we could well document it somewhere.

**Comment by lifang0504** [comment]
the upstream kernel occur Ethernet problem. the eno1 can't active through iot2050setup tool and will call trace, keep output "xmit timeout", eno2 can get IP address but can not ping successful.
[upstreamkernel_eth.txt](https://github.com/siemens/meta-iot2050/files/7118988/upstreamkernel_eth.txt)

File: recipes-kernel/linux/files/patches-5.10/0047-arm64-dts-ti-iot2050-Add-support-for-product-generat.patch [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am6548-iot2050-advanced-pg2.dts
+@@ -0,0 +1,26 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * Copyright (c) Siemens AG, 2018-2021
++ *
++ * Authors:
++ *   Le Jin <le.jin@siemens.com>
++ *   Jan Kiszka <jan.kiszk@siemens.com>
```
[the remark this answers is not in the document]
Comment (jan-kiszka) [reply recipes-kernel/linux/files/patches-5.10/0047-arm64-dts-ti-iot2050-Add-support-for-product-generat.patch:387]: indeed, thanks

**Comment by jan-kiszka** [comment]
@lifang0504, @BaochengSu told me the issue was resolved already, is that correct? Wrong firmware?

**Comment by BaochengSu** [comment]
> ?

Yep, wrong firmware.

**Comment by AsuraZeng** [comment]
As the hardware watchdog driver is not ready.we'd like to use R5 watchdog.
currently R5 watchdog is missing in the DTS, should pick the commit https://github.com/siemens/meta-iot2050/commit/acd739022b020494c0e679af02b8e29f37ef0822 from my branch.

It would be a local patch to reside in our meta repo

**Comment by jan-kiszka** [comment]
Yeah, found this out as well. I'm preparing a proper upstream patch that follows the EVM in this regard (there are significant deltas to your patch).

**Comment by jan-kiszka** [comment]
Updated the queue, moving to 5.10.63, re-adding spidev and also a new commit for the R5F core nodes.

**Pull request merged**